src/dft/backends/cufft/commit.cpp

```diff
@@ -144,19 +144,19 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         const int bwd_dist = static_cast<int>(config_values.bwd_dist);
         std::array<int, max_supported_dims> inembed;
         if (rank == 2) {
-            inembed[1] = config_values.input_strides[1];
+            inembed[1] = config_values.input_strides[1] / istride;
         }
         else if (rank == 3) {
-            inembed[2] = config_values.input_strides[2];
-            inembed[1] = config_values.input_strides[1] / inembed[2];
+            inembed[2] = config_values.input_strides[2] / istride;
+            inembed[1] = (config_values.input_strides[1] / inembed[2]) / istride;
         }
         std::array<int, max_supported_dims> onembed;
         if (rank == 2) {
-            onembed[1] = config_values.output_strides[1];
+            onembed[1] = config_values.output_strides[1] / ostride;
         }
         else if (rank == 3) {
-            onembed[2] = config_values.output_strides[2];
-            onembed[1] = config_values.output_strides[1] / onembed[2];
+            onembed[2] = config_values.output_strides[2] / ostride;
+            onembed[1] = (config_values.output_strides[1] / onembed[2]) / ostride;
```

Comment:
- I don't understand how cuFFT could work if `{i,o}stride` are set as the minimum value as you described. How would the library know which dimension this stride is corresponding to? According to the doc: `The istride and ostride parameters denote the distance between two successive input and output elements in the least significant (that is, the innermost) dimension respectively` so the current implementation makes sense to me. The example you describe runs into the issue in the second point though.
- I was also wondering about this issue. It also seems to me that cuFFT cannot handle all the strides but we need to investigate more. We didn't plan to work on adding more tests for non-default strides now but @t4c1 will be able to look into this after all.
Overall I agree this needs more testing but it seemed to be a step in the right direction to support more cases, like the ones used in https://github.com/oneapi-src/oneMKL/issues/341. Let me know if you prefer that we wait to add more tests and configuration checks before we merge this.

Reply:
- Well, I was not referring to feeding those strides directly into cuFFT as is but my thinking was that a user of the oneAPI project **may** set the `input_strides` and `output_strides` as such one day, and wonder why it doesn't work with cuFFT backend whereas it does with (closed-source) oneMKL backend. If a user were to describe a 2D AxB layout as illustrated above (_i.e._, describing it using apparent column-major ordering for some unknown reason that may make sense to them), it shouldn't make any difference to feed the very same data set to cuFFT but describe it (to cuFFT) as a BxA 2D regular layout (_i.e._, with standard row-major ordering). Any dimension can be done first for c2c FFTs (but it's not as straightforward for r2c/c2r). Whether the oneAPI users set their strides consistently with the concept that the last stride element corresponds indeed to the `least significant` dimension or not is irrelevant, their minimal stride will correspond to the `least significant` dimension's stride in the end;
- The current implementation would not enable it anyway since remainders are simply ignored in the integer divisions, unless I'm mistaken. I think it'd be safer to check for those remainders to be 0 before proceeding and throw if any remainder is not zero.

Reply:
Right I understand, in other words the oneMKL spec lets the user choose which dimension is the least significant. I was not planning to support the cases you describe as the goal was only to support configurations that can directly map from oneMKL to the other backends for now. Tadej's PR will improve this for some cases and throw unsupported when it is not possible or too complex. I'll close this PR now.